web/src/main/java/se/inera/intyg/rehabstod/service/pdl/PdlLogMessageFactoryImpl.java

```diff
@@ -87,6 +93,51 @@ public PdlLogMessage buildLogMessage(LogPatient logPatient,
         return pdlLogMessage;
     }
 
+    @Override
+    public PdlLogMessage buildLogMessage(List<LUCertificate> luCertificateList, LogUser logUser, ActivityType activityType,
+        ResourceType resourceType, Map<String, List<PDLActivityEntry>> storedActivities) {
+        PdlLogMessage pdlLogMessage = getLogMessage(activityType);
+        populateWithCurrentUserAndCareUnit(pdlLogMessage, logUser);
+
+        // Add resources
+        pdlLogMessage.getPdlResourceList().addAll(
+            luCertificateList.stream().filter(distinctByKey(c -> c.getPatient().getId()))
+                .map(c -> buildPdlLogResource(c, logUser, resourceType))
+                .collect(Collectors.toList()));
+
+        // Unset values due to regulations
+        unsetValues(pdlLogMessage);
```

Comment:
I would rename the method to unsetValuesDueToRegulations and remove the comment.